controllers/front/SmailyRssFeed.php

```diff
@@ -40,39 +40,47 @@ public function generateRssFeed()
             htmlspecialchars($baseUrl).'</link><description>Product Feed</description><lastBuildDate>' .
             date("D, d M Y H:i:s") . '</lastBuildDate>';
         foreach ($products as $product) {
-            $link = new Link();
-            $product_url = $link->getProductLink((int)$product['id_product']);
-            $image = Product::getCover((int)$product['id_product']);
+            // Product data by id.
+            $prod = new Product($product['id_product']);
+            // Product name.
+            $name = $product['name'];
+            // Product url.
+            $product_url = $prod->getLink();
+            // Date added.
+            $date_add = $prod->date_add;
+            // Description (short).
+            $description_short = $product['description_short'];
+            // Product image.
+            $image = $prod->getCover($product['id_product']);
             $image = new Image($image['id_image']);
             $product_photo = _PS_BASE_URL_ . _THEME_PROD_DIR_ . $image->getExistingImgPath() . ".jpg";
```

Comment:
Are all images in JPEG format?

Reply:
Changed for dynamic behaviour 33a09cd3a7c158e7bf78d29b7cf63193202e7f33